### PR TITLE
test(suites): traefik3

### PR DESCRIPTION
**Pull request by james-d-elliott**
This adds Traefik v3 to the suites and formally deprecates Traefik v1. 

Closes #7426

### CODE REVIEW

File: internal/suites/example/compose/nginx/backend/docker-compose.yml [hunk 1]
```diff
@@ -3,15 +3,8 @@ services:
...
-      # Traefik 1.x
-      - 'traefik.frontend.rule=Host:admin.example.com,deny.example.com,home.example.com,public.example.com,secure.example.com,singlefactor.example.com'  # yamllint disable-line rule:line-length
-      - 'traefik.frontend.auth.forward.address=https://authelia-backend:9091/api/authz/forward-auth'  # yamllint disable-line rule:line-length
-      - 'traefik.frontend.auth.forward.tls.insecureSkipVerify=true'
-      - 'traefik.frontend.auth.forward.trustForwardHeader=true'
-      - 'traefik.frontend.auth.forward.authResponseHeaders=Authorization,Proxy-Authorization,Remote-User,Remote-Groups,Remote-Email,Remote-Name'
-      # Traefik 2.x
       - 'traefik.enable=true'
-      - 'traefik.http.routers.protectedapps.rule=Host(`admin.example.com`,`deny.example.com`,`home.example.com`,`public.example.com`,`secure.example.com`,`singlefactor.example.com`)'  # yamllint disable-line rule:line-length
+      - 'traefik.http.routers.protectedapps.rule=Host(`admin.example.com`) || Host(`deny.example.com`) || Host(`home.example.com`) || Host(`public.example.com`) || Host(`secure.example.com`) || Host(`singlefactor.example.com`)'  # yamllint disable-line rule:line-length
```
Comment (james-d-elliott) [inline internal/suites/example/compose/nginx/backend/docker-compose.yml:7]: Fix required for Traefik3

File: internal/suites/suite_traefik2_test.go [hunk 2]
```diff
@@ -1,75 +1,15 @@
...
-	s.verifySecretAuthorized(s.T(), s.Context(ctx))
-}
-
 func TestTraefik2Suite(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping suite test in short mode")
 	}
 
-	suite.Run(t, NewTraefik2Suite())
+	suite.Run(t, NewTraefikSuite(traefik2SuiteName))
```
[the remark this answers is not in the document]
Comment (james-d-elliott) [reply internal/suites/suite_traefik2_test.go:14]: This is expected, the Test itself needs to be named like this.

File: internal/suites/Traefik/configuration.yml [hunk 3]
```diff
@@ -5,12 +5,21 @@
 
 certificates_directory: /certs/
 
+identity_validation:
```
Comment (james-d-elliott) [inline internal/suites/Traefik/configuration.yml:8]: The changes in this file are effectively a move from Traefik2 -> Traefik

File: internal/suites/Traefik/docker-compose.yml [hunk 4]
```diff
@@ -4,6 +4,8 @@ services:
     volumes:
       - './Traefik/configuration.yml:/config/configuration.yml'
       - './Traefik/users.yml:/config/users.yml'
+      - './Traefik/favicon.ico:/config/assets/favicon.ico'
```
Comment (james-d-elliott) [inline internal/suites/Traefik/docker-compose.yml:7]: Added due to the rename from Traefik2 -> Traefik, see the below renames.

File: internal/suites/example/compose/oidc-client/docker-compose.yml [hunk 5]
```diff
@@ -12,7 +12,7 @@ services:
     labels:
       - 'traefik.enable=true'
       - 'traefik.http.routers.oidc.rule=Host(`oidc.example.com`)'
-      - 'traefik.http.routers.oidc.priority=150'
+      - 'traefik.http.routers.oidc.priority=1000'
```
Comment (james-d-elliott) [inline internal/suites/example/compose/oidc-client/docker-compose.yml:15]: Fix required for the Traefik3 fix above.

File: internal/suites/Traefik/users.yml
Comment (Crowley723) [inline internal/suites/Traefik/users.yml:1]: At least one hash has changed from default and the yamllint exception was removed.
Was that intended?

File: internal/suites/Traefik/users.yml [hunk 6]
```diff
@@ -1,35 +1,28 @@
...
+        groups: []
+        disabled: false
+    john:
+        password: $argon2id$v=19$m=65536,t=3,p=4$EiVH8kgEtW6mZahOibRbzQ$SzVoiZ1gfK1XbcMhznnsz6W0v8NXLnpdfzEbeQahC4s
+        displayname: John Doe
+        email: john.doe@authelia.com
+        groups:
+            - admins
+            - dev
+        disabled: false
```
Comment (Crowley723) [inline internal/suites/Traefik/users.yml:28]: Lets keep the users.yml standard.
```suggestion
  john:
    displayname: "John Doe"
    password: "$6$rounds=500000$jgiCMRyGXzoqpxS3$w2pJeZnnH8bwW3zzvoMWtTRfQYsHbWbD/hquuQ5vUeIyl9gdwBIt6RWk2S6afBA0DPakbeWgD/4SZPiS0hYtU/"  # yamllint disable-line rule:line-length
    email: john.doe@authelia.com
    groups:
      - admins
      - dev

  harry:
    displayname: "Harry Potter"
    password: "$6$rounds=500000$jgiCMRyGXzoqpxS3$w2pJeZnnH8bwW3zzvoMWtTRfQYsHbWbD/hquuQ5vUeIyl9gdwBIt6RWk2S6afBA0DPakbeWgD/4SZPiS0hYtU/"  # yamllint disable-line rule:line-length
    email: harry.potter@authelia.com
    groups: []

  bob:
    displayname: "Bob Dylan"
    password: "$6$rounds=500000$jgiCMRyGXzoqpxS3$w2pJeZnnH8bwW3zzvoMWtTRfQYsHbWbD/hquuQ5vUeIyl9gdwBIt6RWk2S6afBA0DPakbeWgD/4SZPiS0hYtU/"  # yamllint disable-line rule:line-length
    email: bob.dylan@authelia.com
    groups:
      - dev

  james:
    displayname: "James Dean"
    password: "$6$rounds=500000$jgiCMRyGXzoqpxS3$w2pJeZnnH8bwW3zzvoMWtTRfQYsHbWbD/hquuQ5vUeIyl9gdwBIt6RWk2S6afBA0DPakbeWgD/4SZPiS0hYtU/"  # yamllint disable-line rule:line-length
    email: james.dean@authelia.com
...

```

File: internal/suites/Traefik/users.yml [hunk 7]
```diff
@@ -1,35 +1,28 @@
----
-###############################################################
-#                         Users Database                      #
-###############################################################
-
-# This file can be used if you do not have an LDAP set up.
-
-# List of users
 users:
```
Comment (Crowley723) [inline internal/suites/Traefik/users.yml:1]:
```suggestion
---
###############################################################
#                         Users Database                      #
###############################################################

# This file can be used if you do not have an LDAP set up.

# List of users
users:
```

**Review by Crowley723 — CHANGES_REQUESTED** [review]
I agree with the grammar changes and I think we should try to keep the user.yml files the same. Otherwise it looks good.

**Review by Crowley723 — APPROVED** [review]

**Pull request merged**